pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -373,6 +374,10 @@ def on_validation_epoch_end(self, trainer, pl_module):
         if self.val_progress_bar_id is not None:
             self._update(self.val_progress_bar_id, self.val_batch_idx, self.total_val_batches, visible=False)
 
+    def on_validation_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        if trainer.state.fn == "fit":
```

Comment:
Why not use the Enum?

Reply:
it is sort of like a public hook, and I see we don't use the Enum in other callbacks as well.